app/services/account_builder.rb

```diff
@@ -0,0 +1,93 @@
+# Builds an Account object given params.
+# This class supports extensions from engines via the `build_subclass` method.
+class AccountBuilder
```

Comment:
I like how you've pulled more into this class. I have a few thoughts.

I'm not sure I'm crazy about injecting more methods into this class and dynamically calling them (the `build_nufs_account` etc). I'd actually be slightly more inclined towards creating subclasses and using a template method for `build_subclass`. You could do a factory method here so `AccountBuilder.for(account_class)` that dynamically sets up a `NufsAccountAccountBuilder`--it could default to this `Accountbuilder` if it can't find one by name. That would also give you a place to put the expiration setter as opposed to requiring the Account subclass implement a `set_expires_at!`.

Another thing that might clean this up would be to take the parameters you currently have for `build` and add them to the initializer to set instance variables. That way you're not passing them all over the place. Then you could have a class method to keep a similar public API.

```
def self.build(facility, current_user, owner_user, params)
  new(facility, current_user, owner_user, params).build
end
```